apps/namadillo/src/App/Sidebars/MaspRewardCalculator.tsx

```diff
@@ -0,0 +1,261 @@
+import { Panel } from "@namada/components";
+import { AssetImage } from "App/Transfer/AssetImage";
+import {
+  chainAssetsMapAtom,
+  chainParametersAtom,
+  maspRewardsAtom,
+} from "atoms/chain";
+import { simulateShieldedRewards } from "atoms/staking/services";
+import clsx from "clsx";
+import { useAtomValue } from "jotai";
+import { useEffect, useRef, useState } from "react";
+import { GoChevronDown } from "react-icons/go";
+import { MaspAssetRewards } from "types";
+
+export const MaspRewardCalculator = (): JSX.Element => {
+  const rewards = useAtomValue(maspRewardsAtom);
+  const [amount, setAmount] = useState<string>("");
+  const [selectedAsset, setSelectedAsset] = useState<
+    MaspAssetRewards | undefined
+  >(rewards.data?.[0] || undefined);
+  const [searchTerm, setSearchTerm] = useState<string>("");
+  const [isDropdownOpen, setIsDropdownOpen] = useState<boolean>(false);
+  const [calculatedRewards, setCalculatedRewards] = useState<string>("0.00");
+  const [isCalculating, setIsCalculating] = useState<boolean>(false);
+  const dropdownRef = useRef<HTMLDivElement>(null);
+  const searchInputRef = useRef<HTMLInputElement>(null);
+  const chainParameters = useAtomValue(chainParametersAtom);
+  const chainId = chainParameters.data?.chainId;
+  const chainAssetsMap = useAtomValue(chainAssetsMapAtom);
+
+  // Set initial selected asset when rewards data loads
+  useEffect(() => {
+    if (rewards.data && rewards.data.length > 0 && !selectedAsset) {
+      setSelectedAsset(rewards.data[0]);
+    }
+  }, [rewards.data, selectedAsset]);
+
+  // Close on click outside
+  useEffect(() => {
+    const handleClickOutside = (event: MouseEvent): void => {
+      if (
+        dropdownRef.current &&
+        !dropdownRef.current.contains(event.target as Node)
+      ) {
+        setIsDropdownOpen(false);
+        setSearchTerm("");
+      }
+    };
+
+    document.addEventListener("mousedown", handleClickOutside);
+    return () => document.removeEventListener("mousedown", handleClickOutside);
+  }, []);
+
+  // calculate rewards when amount or asset changse
+  useEffect(() => {
+    const fetchRewards = async (): Promise<void> => {
+      if (!selectedAsset || !amount || !chainId) return;
+
+      const assetAddress = findAssetAddress(
+        selectedAsset.asset.symbol.toLowerCase()
+      );
+      if (!assetAddress) {
+        console.error(
+          "Could not find address for asset:",
+          selectedAsset.asset.symbol
+        );
+        return;
+      }
+
+      setIsCalculating(true);
+      try {
+        const rewardsResult = await simulateShieldedRewards(
+          chainId,
+          assetAddress,
+          amount
```

Comment:
So amount here is direct value from the input field(which is fine!) but `simulateShieldedRewards` expects amount in min(base) denomination. You can use `toBaseAmount` from utils to convert it

Reply:
resolved